projects/kit/components/input-year/input-year.directive.ts

```diff
@@ -0,0 +1,119 @@
+    selector: 'input[tuiInputYear]',
+    providers: [
+        tuiAsControl(TuiInputYearDirective),
+        tuiValueTransformerFrom(TUI_INPUT_YEAR_OPTIONS),
+    ],
+    hostDirectives: [TuiWithTextfield, MaskitoDirective, TuiDropdownAuto],
+    host: {
+        maxlength: '4',
+        inputmode: 'numeric',
+        '[disabled]': 'disabled()',
+        '(click)': 'toggle()',
+        '(input)': 'onInput($event.target.value)',
+    },
+})
+export class TuiInputYearDirective extends TuiControl<number | null> {
+    private readonly options = inject(TUI_INPUT_YEAR_OPTIONS);
+    private readonly textfield = inject(TuiTextfieldDirective);
+    private readonly open = tuiDropdownOpen();
+
+    private readonly initialItem = computed(
+        () => this.value() ?? this.calendar()?.initialItem() ?? null,
+    );
+
+    protected readonly calendar = tuiInjectAuxiliary<TuiCalendarYear>(
+        (x) => x instanceof TuiCalendarYear,
+    );
+
+    protected readonly dropdownEnabled = tuiDropdownEnabled(
+        computed(() => this.interactive()),
+    );
+
+    protected readonly icon = tuiTextfieldIconBinding(TUI_INPUT_YEAR_OPTIONS);
+    protected readonly min = signal(this.options.min.year);
+    protected readonly max = signal(this.options.max.year);
+
+    /**
+     * TODO: move to [value]="value()" after update to Angular 17+
+     * something wrong with change detection on host binding
+     */
+    protected readonly valueEffect = effect(
+        () => this.textfield.value.set(this.value()?.toString() ?? ''),
+        TUI_ALLOW_SIGNAL_WRITES,
+    );
+
+    protected readonly mask = tuiMaskito(
+        computed(() =>
+            maskitoNumberOptionsGenerator({
+                min: this.min(),
+                max: this.max(),
+                thousandSeparator: '',
+            }),
+        ),
+    );
+
+    protected readonly calendarInEffect = effect(() => {
+        const calendar = this.calendar();
+
+        if (calendar) {
+            calendar.initialItemSetter = this.initialItem();
+            calendar.value.set(this.value());
+            calendar.min.set(this.min());
+            calendar.max.set(this.max());
+        }
+    }, TUI_ALLOW_SIGNAL_WRITES);
+
+    protected readonly calendarOutEffect = effect((onCleanup) => {
+        const subscription = this.calendar()?.yearClick.subscribe((year) => {
+            this.onChange(year);
+            this.open.set(false);
+        });
+
+        onCleanup(() => subscription?.unsubscribe());
+    });
+
+    // TODO(v5): use signal inputs
+    @Input('min')
+    public set minSetter(x: number) {
+        this.min.set(x);
+    }
+
+    // TODO(v5): use signal inputs
+    @Input('max')
+    public set maxSetter(x: number) {
+        this.max.set(x);
+    }
+
+    protected toggle(): void {
+        if (this.interactive()) {
+            this.open.update((x) => !x);
+        }
+    }
+
+    protected onInput(value: string): void {
+        this.onChange(Number(value) || 0);
```

Comment:
- Empty textfield – `null`
- `0` – `0`

The same logic as for `InputNumber`
https://github.com/taiga-family/taiga-ui/blob/e39a857a505ae4312cc3a255e156c15f8750dc46/projects/kit/components/input-number/input-number.directive.ts#L66-L69

Reply:
fixed